docs/DeveloperGuide.md

```diff
@@ -258,85 +260,182 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+    * 1c1. System shows an error message.<br>
+      Use case ends.
+
+* 1d. There is a clash of events in an upcoming day.
+    * 1d1. System shows an error message.<br>
+      Use case ends.
 
 </details>
 
 <details open markdown="block">
-<summary markdown="span">**Use case: Undo a delete**</summary>
+<summary markdown="span">**Use case: Edit an event**</summary>
 
 **MSS**
-1. User requests to undo the recent delete
-2. System restores the most recent delete  
-    Use case ends.
+
+1. User requests to edit an event
+2. System edits the event<br>
+   Use case ends.
 
 **Extensions**
-* 1a. New event added duplicate of recent deleted event.
-    * 1a1. System shows an error message.  
+
+* 1a. The event already exist.
+    * 1a1. System shows an error message.<br>
+      Use case ends.
+
+* 1b. The given format invalid.
+    * 1b1. System shows an error message.<br>
       Use case ends.
-* 2a. The list is empty.  
-  Use case ends.
 
 </details>
 
+<details open markdown="block">
+<summary markdown="span">**Use case: Delete an event**</summary>
+
+**MSS**
+
+1. User requests to delete an event/ events
+2. System deletes the event(s)<br>
+   Use case ends.
+
+**Extensions**
+
+* 1a. The given index/ indices is invalid.
+    * 1a1. System shows an error message.<br>
+      Use case ends.
+
+* 1b. The given format invalid.
+    * 1b1. System shows an error message.<br>
+      Use case ends.
+
+</details>
 
 <details open markdown="block">
-<summary markdown="span">**Use case: Find event by name**</summary>
+<summary markdown="span">**Use case: Find an event by name**</summary>
 
 **MSS**
+
 1. User requests to find an event by name
-2. System shows a list of events matching the name  
+2. System finds events with matching name
+3. System displays events with matching name<br>
+   Use case ends.
+
+**Extensions**
+
+* 1a. The given event format invalid.
+    * 1a1. System shows an error message.<br>
+      Use case ends.
+
+</details>
+
+<details open markdown="block">
+<summary markdown="span">**Use case: Show next event(s) by name**</summary>
+
+**MSS**
+
+1. User requests to show next x number of event(s)
+2. System displays the next x number of events in chronological order<br>
    Use case ends.
 
 **Extensions**
-* 1a. The list is empty.  
-  Use case ends.
+
+* 1a. The given event format invalid.
+    * 1a1. System shows an error message.<br>
+      Use case ends.
 
 </details>
 
 <details open markdown="block">
-<summary markdown="span">**Use case: Find next event**</summary>
+<summary markdown="span">**Use case: Undo a delete**</summary>
 
 **MSS**
-1. User requests for next upcoming event
-2. System shows the next upcoming event  
+1. User requests to undo the recent delete
+2. System restores the most recent delete<br>
    Use case ends.
 
 **Extensions**
-* 1a. The list is empty.  
-  Use case ends.
+* 1a. New event added duplicate of recent deleted event.
+    * 1a1. System shows an error message.<br>
+      Use case ends.
```

Comment:
Can you explain this to me? How does the conflict happens?
If you delete an event, and attempts to undo the delete, you are effectively adding back the event to its original timeslot. This timeslot is guaranteed to be empty and have no conflicts with any existing events, since prior to delete, it (should) had no conflict with any existing events.

Reply:
You are right.

Previously i was thinking of 
1)Delete event1
2)Add event1
3) Undo -> clash

But I realise it is wrong. I will remove the extension.

Reply:
Extension 1a: duplicate for undo removed.